src/main/java/fish/payara/extras/upgrade/BaseUpgradeCommand.java

```diff
@@ -224,21 +224,24 @@ private ServiceLocator createServiceLocator() throws CommandException {
     protected boolean reinstallSSHNode(Node node) {
         LOGGER.log(Level.INFO, "Reinstalling SSH node {0}", new Object[]{node.getName()});
         ArrayList<String> command = new ArrayList<>();
+        SshConnector sshConnector = node.getSshConnector();
+        SshAuth sshAuth = sshConnector.getSshAuth();
         command.add(SystemPropertyConstants.getAdminScriptLocation(glassfishDir));
         command.add("--interactive=false");
-        command.add("--passwordfile");
-        command.add("-");
+        if (!ok(sshAuth.getKeyfile())) {
```

Comment:
```suggestion
        if (ok(sshAuth.getPassword())) {
```

Tested with all nodes upgrading successfully against a domain containing:
* Node with no auth (so using default of `~/.ssh/id_rsa` key file)
  * Results in command: `Executing command: [/home/andrew/Downloads/FISH-6240/payara5/glassfish/lib/nadmin, --interactive=false, install-node-ssh, --installdir, /home/andrew/Downloads/FISH-6240, --force, --sshport, 22, --sshuser, andrew, opensuse-mini-2]`
* Node with password auth
  * Results in command: `Executing command: [/home/andrew/Downloads/FISH-6240/payara5/glassfish/lib/nadmin, --interactive=false, --passwordfile, -, install-node-ssh, --installdir, /home/andrew/Downloads/FISH-6240-Password, --force, --sshport, 22, --sshuser, andrew, opensuse-mini-2]`
* Node with defined key
  * Results in command: `Executing command: [/home/andrew/Downloads/FISH-6240/payara5/glassfish/lib/nadmin, --interactive=false, install-node-ssh, --installdir, /home/andrew/Downloads/FISH-6240-Key, --force, --sshport, 22, --sshuser, andrew, --sshkeyfile, /home/andrew/.ssh/id_rsa, opensuse-mini-2]`
* Node with both defined key AND password
  * Results in command: `Executing command: [/home/andrew/Downloads/FISH-6240/payara5/glassfish/lib/nadmin, --interactive=false, --passwordfile, -, install-node-ssh, --installdir, /home/andrew/Downloads/FISH-6240-Both, --force, --sshport, 22, --sshuser, andrew, --sshkeyfile, /home/andrew/.ssh/id_rsa, opensuse-mini-2]`